src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -193,18 +193,48 @@ def shallow_copy(self):
         are new (symbols added to the new symbol table will not be added in
         the original but the existing objects are still the same).
 
-        :returns: a copy of this symbol table.
+        :returns: a shallow copy of this symbol table.
         :rtype: :py:class:`psyclone.psyir.symbols.SymbolTable`
 
         '''
         # pylint: disable=protected-access
         new_st = SymbolTable()
-        new_st._symbols = copy(self._symbols)
-        new_st._argument_list = copy(self._argument_list)
-        new_st._tags = copy(self._tags)
+        new_st._symbols = copy.copy(self._symbols)
+        new_st._argument_list = copy.copy(self._argument_list)
+        new_st._tags = copy.copy(self._tags)
         new_st._node = self.node
         return new_st
 
+    def deep_copy(self):
+        '''Create a copy of the symbol table where the top-level containers
```

Comment:
Perhaps: "...containers (lists and dicts) are new..." as I found this a bit hard to understand at first.

Reply:
I had "container" because it was the same in the shallow copy docstring, but you are right. I updated both shallow and deep copy docstring to hopefully a clearer description.

Reply:
Much clearer, thanks.